src/widgets/Ruler.h

```diff
@@ -62,6 +63,13 @@ class AUDACITY_DLL_API Ruler {
    // (at the center of the pixel, in both cases)
    void SetRange(double min, double max, double hiddenMin, double hiddenMax);
 
+   // Set the kind of updater the ruler will use
+   // (Linear, Logarithmic, Custom, etc.)
+   void SetUpdater(std::unique_ptr<Updater> pUpdater);
+
+   // An overload to replace SetUseZoomInfo
+   void SetUpdater(std::unique_ptr<Updater> pUpdater, int leftOffset);
```

Comment:
I see two overloads of `SetUpdater`.  Can they be collapsed into one, by use of a default argument for leftOffset?

Yes:  you might pass `int*` not `int` and default the argument to `nullptr`

But better than that -- avoiding funny looking function declarations that might leave the reviewer with doubts about  lifetime questions -- you can make the intent even clearer by passing `std::optional<int>` around, by-value.

Reply:
I have never heard of `std::optional<>`. I will look into it.